Main i18n omissions

j-dornbusch opened the pull request:
Fixes some omissions in internationalization
@cjcolvar This PR includes the contents of the previous one without the automatic translation script based on Yandex : however I continue to use it, to avoid the Google developer account.
@samvera/hyrax-code-reviewers

j-dornbusch commented:
@jcoyne I dont really undertand why a test is looking for "en.hyrax.base.show_actions**.html.erb**.analytics". Internationalization keys do not include file extensions ? What am I missing ?

jcoyne commented:
@j-dornbusch I have no idea. That looked really odd to me too.

j-dornbusch commented:
@jcoyne 
This line in the test : 
https://github.com/samvera/hyrax/blob/37a08e384bb9b7dbc8859dc3d126e533e534814d/spec/views/hyrax/base/_show_actions.html.erb_spec.rb#L18
seems suspicious to me.

j-dornbusch commented:
This syntax is present only in this spec and in this other one : https://github.com/samvera/hyrax/blob/master/spec/views/hyrax/file_sets/_single_use_links.html.erb_spec.rb

cjcolvar commented:
FWIW I verified locally on master that those two spec files still pass after removing `.html.erb` from the `render` lines.  So I think you should go ahead and make that change in this PR.

j-dornbusch commented:
@cjcolvar Searching with regex `render '[^']*.html.erb'` in spec files gives 13 results, I eliminated 9 of them.

j-dornbusch commented:
Searching with `render template: '[^']*.html.erb'` gives 2 occurences, I eliminated both.

j-dornbusch commented:
The rake task for alphabetical sorting had not been applied before this PR. It moved some keys and apparently it does not preserve double line breaks.

j-dornbusch commented:
@cjcolvar I corrected everything I could but the last failure ("Circular dependency detected while autoloading constant Hyrax::Actors::OrderedMembersActor") has really nothing to do with the changes I made...

cjcolvar commented:
I was able to fix the circular dependency issue locally by using `Rails.application.reloader.wrap` within the thread (thanks to [this issue](https://github.com/rails/rails/issues/30350) for the hint).  Here's a diff of what I did.  I could make a Hyrax PR with this or you could just include it in your PR whichever is easiest for you.

```
diff --git a/spec/actors/hyrax/actors/create_with_remote_files_ordered_members_actor_spec.rb b/spec/actors/hyrax/actors/create_with_remote_files_ordered_members_actor_spec.rb
index a33f2728a..e0984fdbd 100644
--- a/spec/actors/hyrax/actors/create_with_remote_files_ordered_members_actor_spec.rb
+++ b/spec/actors/hyrax/actors/create_with_remote_files_ordered_members_actor_spec.rb
@@ -43,8 +43,10 @@ RSpec.describe Hyrax::Actors::CreateWithRemoteFilesOrderedMembersActor do
       expect(ImportUrlJob).to receive(:perform_later).with(FileSet, null_operation, {}).exactly(4).times
       threads = environments.collect do |env|
         Thread.new do
-          expect(actor.create(env)).to be true
-          expect(env.curation_concern.ordered_members.to_a.collect(&:label)).to eq(filenames)
+          Rails.application.reloader.wrap do
+            expect(actor.create(env)).to be true
+            expect(env.curation_concern.ordered_members.to_a.collect(&:label)).to eq(filenames)
+          end
         end
       end
       threads.each(&:join)
```

j-dornbusch commented:
Thanks a lot @cjcolvar. I added your path to my PR (with a wrong commit message, sorry).
Now I am able to run the tests locally again: we had a problem with [this issue](https://github.com/samvera/hyrax/issues/3536) but we found a workaround.

j-dornbusch commented:
@cjcolvar My IDE said it: with your fix, the example is now too long.... Rubocop does not agree...

cjcolvar commented:
Yeah, I'm surprised that hasn't been fixed yet in master so thanks for writing it up! The known good workaround for now is upgrade engine_cart to 2.2.

j-dornbusch commented:
@cjcolvar I did not manage to use your fix. Maybe you could pass it yourself and I will rebase my branch?

cjcolvar commented:
I think those two expectations need to be together in the thread.  How about declaring the threads in a `let` then the body of the example becomes:
```
expect(ImportUrlJob).to receive(:perform_later).with(FileSet, null_operation, {}).exactly(4).times
threads.each(&:join)
```
That or you could make a rubocop exception or disable the cop inline around the test (like [this](https://github.com/samvera/hyrax/blob/master/spec/services/hyrax/adapters/nesting_index_adapter_spec.rb#L150-L183))

cjcolvar commented:
Sure, I could do that.

cjcolvar commented:
@j-dornbusch Don't worry about the one failing travis build since circleci is now the standard for mergeability.
I think this could be merged once it has been reviewed and doesn't need to be rebased with #3537 since the tests passed this time.
@jcoyne are you able to review this PR again?

jmfrenzel commented on config/locales/hyrax.de.yml:
```diff
@@ -430,7 +427,16 @@ de:
+        relationships: Beziehungen
+      show_actions:
+        analytics: Analytics
+        attach_child: Befestigen Kind
```
Please change this into :
"Untergeordnete Arbeit hinzufügen"

jmfrenzel commented on config/locales/hyrax.de.yml:
```diff
@@ -430,7 +427,16 @@ de:
+        attach_child: Befestigen Kind
+        delete: Löschen
+        edit: Bearbeiten
+        feature: Feature
```
Please change this into :
"Hervorheben"

jmfrenzel commented on config/locales/hyrax.de.yml:
```diff
@@ -430,7 +427,16 @@ de:
+        delete: Löschen
+        edit: Bearbeiten
+        feature: Feature
+        unfeature: Unfeature
```
Please change this into :
"Hervorhebung entfernen"

jmfrenzel commented on config/locales/hyrax.de.yml:
```diff
@@ -430,7 +427,16 @@ de:
         definitions_html: "<strong>Ansicht / Download: Auf</strong> diese Datei (sowohl Inhalte als auch Metadaten) kann innerhalb von %{application_name} zugegriffen werden. <br /> <strong>Bearbeiten:</strong> Diese Datei (sowohl Inhalte als auch Metadaten) kann bearbeitet werden. Sie dürfen diese Berechtigung nur Benutzern und / oder Gruppen von %{institution_name} erteilen."
         share_with_html: Sie können den Zugriff "Ansicht / Download" oder "Bearbeiten" für bestimmte Benutzer und / oder Gruppen auf Dateien gewähren. Geben Sie nacheinander einen gültigen %{account_name} ein, wählen Sie die Zugriffsebene für diesen Benutzer und klicken Sie auf + Hinzufügen.
       show:
+        items: Artikel
```
Please change this into :
"Objekte"

jmfrenzel commented on config/locales/hyrax.de.yml:
```diff
@@ -364,15 +364,10 @@ de:
           actions: Aktion
           title: Arbeitstitel
       form_files:
+        add_files: Hinzufügen von Dateien
```
Please change this into:
"Dateien hinzufügen"

j-dornbusch commented:
Sorry @jcoyne you just validated the PR but I had to integrate the improvements of @jmfrenzel

cjcolvar commented:
I reran the circleci build but it still failed on the unrelated test that is fixed by #3537 which is already in master.  I wouldn't hold up this PR over a rebase.

j-dornbusch commented:
@jcoyne @cjcolvar I rebased and cleaned history. I had to remove Rubocop "disable" directives from create_with_remote_files_ordered_members_actor_spec.rb.

j-dornbusch commented:
@cjcolvar I took exactly your version of `create_with_remote_files_ordered_members_actor_spec.rb` but circleci test failed although the PR doesn't change that file.

cjcolvar commented:
That's odd.  I'm not able to recreate that locally.  I'm going to try restarting the build.

j-dornbusch commented:
@cjcolvar  Restarting it solved the problem. There's something unclear, but the important thing is that it works at the end.

cjcolvar commented:
🎉 Thanks for all your work this PR!  It definitely had more tricky spots than a usual one.
